### ISSUE TITLE
std.fmt.parseFloat treats "." as valid float.

### ISSUE THREAD

**Issue by hexaflex** [ziglang/zig#11713]
### Zig Version

0.10.0-dev.1933+5f2d0d414

### Steps to Reproduce

```zig
const std = @import("std");

test "" {
    const f = try std.fmt.parseFloat(f64, ".");
    try std.testing.expect(f == 0);
}
```

> $ zig build test

### Expected Behavior

I would expect this test to fail at the `parseFloat` call.

### Actual Behavior

> $ All 1 tests passed.

**Comment by arBmind** [comment]
I added this test case in #11727 but could not reproduce this issue.

```ps
C:\Zig\Tests> C:\Zig\repo\stage1\bin\zig.exe test float.zig

Test [1/1] testTest [1/1] test ""... FAIL (InvalidCharacter)
FAIL (InvalidCharacter)
C:\Zig\repo\lib\std\fmt\parse_float\parse_float.zig:31:9: 0x7ff65c742161 in td.fmt.parse_float.parse_float.parseFloat (test.obj)
        return parse.parseInfOrNan(T, s[i..], negative) orelse error.InvalidCharacter;
        ^
C:\Zig\Tests\float.zig:4:15: 0x7ff65c74176c in est "" (test.obj)
    const f = try std.fmt.parseFloat(f64, ".");
              ^
0 passed; 0 skipped; 1 failed.
error: the following test command failed with exit code 1:
```

**Comment by tiehuis** [comment]
Your zig version is slightly out of date. A new float-parser has been recently implemented in master which fixes this issue: see #11566.

The following is true as of current; `try testing.expectError(error.InvalidCharacter, parseFloat(f64, "."));`.
